### PR TITLE
fix(sourcemap): Improve scope name resolution

### PR DESCRIPTION
**Pull request by loewenheim**
We have seen cases in Flutter where a scope name cannot be resolved, but the scope's closing brace has the correct name attached. Therefore, if we can't resolve a scope name the straightforward way, we try to grab the name off the scope's last token.

### CODE REVIEW

File: symbolic-sourcemapcache/src/writer.rs [hunk 1]
```diff
@@ -174,6 +196,9 @@ impl SourceMapCacheWriter {
                 };
 
                 let name = token.get_name();
+                if token.get_name_id() == 8638 {
+                    println!("({min_line}, {min_col})");
+                }
```
Comment (Swatinem) [inline symbolic-sourcemapcache/src/writer.rs:201]:
```suggestion
```

:-)

File: symbolic-sourcemapcache/src/writer.rs [hunk 2]
```diff
@@ -71,14 +71,36 @@ impl SourceMapCacheWriter {
...
+                    // A hack specifically for Flutter. If the resolved scope name is the same as the original name,
+                    // that indicates that we probably couldn't resolve the scope. In that case, we find the name
+                    // at the very end of the scope, if it exists, and use it instead of the "conventionally"
+                    // resolved scope.
+                    if orig_name == resolved_name {
+                        let name_at_end_of_scope =
+                            ctx.offset_to_position(range.end - 1).and_then(|sp| {
+                                let token = sm.lookup_token(sp.line, sp.column);
+                                token
+                                    .filter(|t| t.get_dst() == (sp.line, sp.column))
```
Comment (Swatinem) [inline symbolic-sourcemapcache/src/writer.rs:93]: We are looking for an exact token match at the closing `}` which is good. However in testing, we figured out that this trailing token is exactly 1 byte wide.
We could further filter this down to make sure that the token at `range.end` is a different token. I wonder how much that would change things though.

**Review by Swatinem — COMMENTED** [review]
it might be worth extracting that into its own `fn try_resolve_closing_name(ctx, sm, range)` or some such

File: symbolic-sourcemapcache/src/writer.rs (hunk 1 above)
Comment (loewenheim) [reply symbolic-sourcemapcache/src/writer.rs:201]: 🙈

File: symbolic-sourcemapcache/src/writer.rs (hunk 2 above)
Comment (loewenheim) [reply symbolic-sourcemapcache/src/writer.rs]: Would it make sense to check for a literal closing brace?

**Review by Swatinem — APPROVED** [review]

**Pull request merged**